implementations/rust/ockam/ockam_command/src/tcp/outlet/show.rs

```diff
@@ -48,12 +60,25 @@ pub async fn run_impl(
     let node = BackgroundNode::create(&ctx, &opts.state, &node_name).await?;
     let outlet_status: OutletStatus = node.ask(&ctx, make_api_request(cmd)?).await?;
```

Comment:
Since the `OutletStatus` struct already implements the `Serialize/Deserialize` traits, you can leverage the `Terminal` struct to print the status in multiple formats. [Here is an example](https://github.com/build-trust/ockam/blob/f9d722579eb543d0f327e34bccaa4bd7b3fc1555/implementations/rust/ockam/ockam_command/src/tcp/inlet/show.rs#L59-L71) of how to write the output of a command properly. You'll also need to refactor a little bit the `plain` format and you can use `formatdoc` to build it, as in the preivous example.

Reply:
Thanks! Updated the code and the image in the description

If you'd prefer I have serde convert `OutletStatus` to JSON directly, these are additional fields that would be present:

![image](https://github.com/build-trust/ockam/assets/772507/0c34c959-ee6b-4d0a-8eca-272596dc8434)

Reply:
Oh my bad! I forgot about those ugly looking fields. To make this look good, you need to specify how to serde the `payload` and `worker_addr` fields.

The `payload` field in the `OutletStatus` struct could be defined as `#[serde(skip_serializing_if = "Option::is_none")]`.

Since the [`Address` type](https://github.com/build-trust/ockam/blob/19a7304d20e6a281cc8668a4dd915ec958115595/implementations/rust/ockam/ockam_core/src/routing/address.rs#L36) implements `From<String>`, you could use `#[serde(from = "String", into = "String")]` in the `Address` type. That way the `worker_addr` field will look as you would expect.

If you go this way, please add the following test at the end of the `address.rs` file

```rust
#[test]
fn serde_addr() {
    let addr = Address::from_string("1#remote_friend");

    let serialized = serde_json::to_string(&addr).unwrap();
    assert_eq!(serialized, "\"1#remote_friend\"".to_string());

    let deserialized: Address = serde_json::from_str(&serialized).unwrap();
    assert_eq!(deserialized, addr);
}
```

Reply:
Sounds good, I'll switch.